### PR TITLE
optimise dyups module  mutex compete

### PR DESCRIPTION
**Pull request by zjd87**

### CODE REVIEW

File: modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c [hunk 1]
```diff
@@ -568,7 +568,15 @@ ngx_http_dyups_init_process(ngx_cycle_t *cycle)
     timer->log = cycle->log;
     timer->data = dmcf;
 
-    ngx_add_timer(timer, dmcf->read_msg_timeout);
+    /*
+     * when init process, break up timer, in case of shpool->mutex compete
```
Comment (chobits) [inline modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c:572]: Can you describe in detail the problem you encountered?

Comment (zjd87) [reply modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c:572]: if every worker process  start timer at the same time,  `ngx_shmtx_lock` (in timer's handler)  may  mutex compete in all processes ,  there ere hang in the process of not getting mutex.  When  break up the start time of timer ,  avoid  mutex compete in all  processes.

File: modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c [hunk 2]
```diff
@@ -568,7 +568,15 @@ ngx_http_dyups_init_process(ngx_cycle_t *cycle)
...
+     */
+    srandom(ngx_pid);
```
Comment (chobits) [inline modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c:574]: srandom is already initilized in , so we do not need it here

```
ngx_worker_process_init
  -> srandom
  -> init_process (ngx_http_dyups_init_process)
```

**Review by zjd87 — COMMENTED** [review]
srandom(seed) has removed

File: modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c [hunk 3]
```diff
@@ -568,7 +568,15 @@ ngx_http_dyups_init_process(ngx_cycle_t *cycle)
...
+
+    delay = dmcf->read_msg_timeout > 1000 ? dmcf->read_msg_timeout : 1000;
+    t = ngx_random() % delay;
```
Comment (chobits) [inline modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c:575]: use one variable for t and delay, they are the same usage. So we can only declare one var

Comment (zjd87) [reply modules/ngx_http_upstream_dyups_module/ngx_http_dyups_module.c]: done

**Review by chobits — APPROVED** [review]

**Comment by chobits** [comment]
Thanks, have merged with minor fix on the commit message

**Pull request merged**